Generic Sensor: Add functionality tests for OrientationSensor

Honry opened the pull request:
<!-- Reviewable:start -->

<!-- Reviewable:end -->

Honry committed:
Generic Sensor: Add functionality tests for AbsoluteOrientationSensor

tobie commented:
Bot ping

tobie commented:
Bot ping

tobie commented:
Bot ping

w3c-bots commented:
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/258414815)

# Firefox (nightly) #

w3c-bots commented:
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/258414816)

# Sauce (safari) #

w3c-bots commented:
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/258414822)

# Chrome (unstable) #

w3c-bots commented:
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/258414823)

# Sauce (MicrosoftEdge) #

Honry committed:
Add RelativeOrientationSensor

Honry committed:
Fix TRAILING WHITESPACE

Honry commented:
@alexshalamov, @pozdnyakov, PTAL.

alexshalamov commented on orientation-sensor/OrientationSensor.https.html:
```diff
@@ -0,0 +1,127 @@
+
+    let mat_64 = new Float64Array(16);
+    sensor.populateMatrix(mat_64);
+    assert_array_approx_equals(mat_64, mat_expect, double_precision);
```
why not to use assert_array_equals from testharness? DomMatrix and Float64Array would not lose precision.

Honry replied:
We found the test result is sometime failed due to lose precision issue in Float64Array and thought it was same as Float32Array. I will fixed it and file bug if the failure occurs again.

alexshalamov commented on orientation-sensor/OrientationSensor.https.html:
```diff
@@ -0,0 +1,127 @@
+    // Works for all supported types.
+    let mat_32 = new Float32Array(16);
+    sensor.populateMatrix(mat_32);
+    assert_array_approx_equals(mat_32, mat_expect, float_precision);
```
Looks like WebAudio has similar helper, maybe it needs to be moved to generic testharness?

Honry replied:
Indeed, I will try to add it to testharness and back to update the test once it is accepted.

Honry replied:
@alexshalamov, this assertion has been accepted in testharness.js, see https://github.com/w3c/web-platform-tests/pull/6682

Honry committed:
address @alexshalamov's comments

 - Do not use approx for DOMMatrix and Float64Array
 - Convert DOMMatrix to Float64Array for comparision

Honry committed:
Remove function assert_array_approx_equals as it is now in generic testharness.js

Honry commented:
@alexshalamov, thanks for your review, I will merge it directly.
